### ISSUE TITLE
Running autoconf instructions fail

### ISSUE THREAD

**Issue by Taneb** [metamath/metamath-exe#3]
Running the `autoconf` instructions listed in the README (`autoreconf -i && ./configure && make`) from a checkout of this repository fail with:

```
make[1]: *** No rule to make target 'big-unifier.mm', needed by 'all-am'. Stop.
make[1]: Leaving directory '/home/nathan/Documents/metamath-canon'
Makefile:366: recipe for target 'all' failed
make: *** [all] Error 2
```

It seems like the file `big-unifier.mm` (and possibly others) were omitted from this distribution, causing `make` to fail.

**Comment by digama0** [comment]
I didn't even know we had a makefile... the instructions on the readme just say to run something like `gcc m*.c` and I think that works.

The `*.mm` files are not needed for building the program, they are only there for testing purposes as example inputs, and this repo apparently doesn't have them because they are tracked in a separate repo (https://github.com/metamath/set.mm/). You could try removing the `dist_pkgdata_DATA` line from the makefile? (I'm no expert on make usage though, so take my advice with a grain of salt.)

**Comment by Taneb** [comment]
@digama0 
```
As an alternative, if you have autoconf, automake, and a C compiler, you
can compile with the command "autoreconf -i && ./configure && make".
This "autoconf" approach automatically finds your compiler and its
options, and configure takes the usual options (e.g., "--prefix=/usr").
The resulting executable will typically be faster because it will check for
and enable available optimizations; tests found that the "improve"
command ran 28% faster on gcc when using an autoconf-generated "configure".
You can again type "./metamath set.mm" to run.  After "make" you may
install it elsewhere using "sudo make install" (note that this installs
".mm" files in the pkgdata directory, by default
"/usr/local/share/metamath/").  If you install it this way, you can then
run metamath as "metamath /usr/share/metamath/set.mm", copy set.mm
locally (cp /usr/share/metamath/set.mm . ; metamath set.mm), or run
metamath and type:  read "/usr/share/metamath/set.mm" (note that inside
metamath, filenames containing "/" must be quoted).
```

**Comment by digama0** [comment]
Yes I'm aware that there are alternatives listed. They likely date back to the original distribution of metamath.exe that included these files. Again, does it work if you remove those lines from the makefile? I don't think the readme should be updated until we know what works and what doesn't. It's clearly not been exercised very much.

**Comment by Taneb** [comment]
Yes, removing those lines does work

**Comment by david-a-wheeler** [comment]
On September 30, 2019 6:51:01 AM EDT, Mario Carneiro <notifications@github.com> wrote:
>I didn't even know we had a makefile... the instructions on the readme
>just say to run something like `gcc m*.c` and I think that works.

The autoconf tools execution is part of the Travis checks that are run on every commit to set.mm. The .travis.yml invokes scripts/build-metamath which invokes auyoconf.

However, in that circumstance the .mm files that were mentioned already are available.

I use autoconf also, but it can I have the files so my tests didn't notice the problem.

Using the autoconf version produces a significantly faster version of metamath.exe.



--- David A.Wheeler

**Comment by Taneb** [comment]
Fixed by #6
